default/python/AI/PolicyAI.py

```diff
@@ -164,14 +167,14 @@ def _determine_influence_priority(self, new_production):
         debug("Setting influence priority to %.1f", priority)
         self._aistate.set_priority(PriorityType.RESOURCE_INFLUENCE, priority)
 
-    def _get_infl_prod(self, update=False):
+    def _get_infl_prod(self, update=False) -> float:
         """Get / Update IP production."""
         if update:
             self._universe.updateMeterEstimates(self._populated_planet_ids)
             fo.updateResourcePools()
         return self._empire.resourceProduction(fo.resourceType.influence)
 
-    def _can_adopt(self, name, replace_other=None):
+    def _can_adopt(self, name: str, replace_other: str = None) -> bool:
```

Comment:
Techically `replace_other: str` is not str but it's [Optional](https://docs.python.org/3.6/library/typing.html#typing.Optional)[str]. Which is `Union[X, None]`